alpha/lib/model/entry.php

```diff
@@ -2132,14 +2132,22 @@ public function setEntitledPusersEdit($v)
 
 			if ($kuser->getStatus() === KuserStatus::BLOCKED && !$this->isEntitledKuserEdit($kuser->getId()))
 			{
-				throw new kCoreException('Cannot add a blocked user', kCoreException::INVALID_USER_ID);
+				$this->blockedUserFlow($kuser->getId(), $parentEntry);
 			}
 
 			$entitledUserPuserEdit[$kuser->getId()] = $kuser->getPuserId();
 		}
 
 		$this->putInCustomData ( "entitledUserPuserEdit" , serialize($entitledUserPuserEdit) );
 	}
+
+	protected function blockedUserFlow($kuserId, $parentEntry = null)
```

Comment:
i would make this a boolean function and use it in the set entitled view/edit.. then throw the error from there.
its less common to create a function that if -> throws
then you can also give it a more meanningfull name like isBlockedUserAllowed